### DISCUSSION TITLE
Separating vault prefix, username and vault suffix in ess locale (messages)

### DISCUSSION THREAD

**Discussion by MotikaCraft-TheJoshua** [EssentialsX/Essentials#4514] [Ideas and Feature Suggestions] [answered]
Change of messages:
```
teleportRequest=§c{0}§6 has requested to teleport to you.
teleportHereRequest=§c{0}§6 has requested that you teleport to them.
```
To:
```
teleportRequest=§c{0} {1} {2}§6 has requested to teleport to you.
teleportHereRequest=§c{0} {1} {2}§6 has requested that you teleport to them.
```

Where:
`{0}` is vault prefix
`{1}` is username
`{2}` is vault suffix

And if vault is not installed, then `{0}` and `{2}` is "" (nothing)

**Comment by MotikaCraft-TheJoshua** [comment]
I also would like to say that this would be cool if you make it so that this can be done for any other messages in ess locale

**Comment by JRoy** [comment, marked as answer]
This isn't practical for the current Essentials translation system, if we separated the prefix/suffix here it wouldn't make sense to not do it for other messages, which is something we will not be doing.

**Reply by MotikaCraft-TheJoshua** [reply]
> This isn't practical for the current Essentials translation system, if we separated the prefix/suffix here it wouldn't make sense to not do it for other messages, which is something we will not be doing.

then why is there an option everywhere else other than locale file to put prefix/suffix/username wherever you want? like for example in essentials chat section

**Reply by JRoy** [reply]
> > This isn't practical for the current Essentials translation system, if we separated the prefix/suffix here it wouldn't make sense to not do it for other messages, which is something we will not be doing.
> 
> then why is there an option everywhere else other than locale file to put prefix/suffix/username wherever you want? like for example in essentials chat section

If you'd like to change if essentials places the prefix/suffix in the displayname itself, you can do that but what you're asking for is different. In the messages you're referencing, `{0}` is just the user's displayname, we're not adding in the prefix/suffix in each message manually.

**Reply by MotikaCraft-TheJoshua** [reply]
> > > This isn't practical for the current Essentials translation system, if we separated the prefix/suffix here it wouldn't make sense to not do it for other messages, which is something we will not be doing.
> > 
> > 
> > then why is there an option everywhere else other than locale file to put prefix/suffix/username wherever you want? like for example in essentials chat section
> 
> If you'd like to change if essentials places the prefix/suffix in the displayname itself, you can do that but what you're asking for is different. In the messages you're referencing, `{0}` is just the user's displayname, we're not adding in the prefix/suffix in each message manually.

can i make that {0} will show username instead of displayname? is there such ess option in config?
